File: movieGenre.py
class Movie:
    def __init__(self):
        self.uuid = ""
        self.title = ""
        self.year = ""
        self.genres = ""
        self.related = []

    def addRelatedMovie(self, m):
        if (type(m) is Movie):
            self.related.append(m)
            return True
        return False

    def setGenre(self, g):
        if(type(g) is Genre):
            self.genres = g
            g.addMovie(self)
            return True
        return False

class Genre:
    def __init__(self):
        self.name = ""
        self.movies = []

    def addMovie(self, m):
        if(type(m) is Movie):
            self.movies.append(m)
            return True
        return False

File: movieGenre.js
//Question 1, 2, 3, 4
function Movie(){
    this.uuid = "";
    this.title = "";
    this.year = "";
    this.genres = "";
    this.related = [];
    this.addRelatedMovie = function(m){
        if (m instanceof Movie){  
            this.related.push(m);
            return true;
        }
        return false;
    }
    this.setGenre = function(g){
        if (g instanceof Genre){
            this.genre = g;
            g.addMovie(this);
            return True;
        }
        return false;
    }
    this.setAttributes = function(uuid, title, year, genres, related){
        this.uuid = uuid;
        this.title = title;
        this.year = year;
        this.genres = genres;
        this.related = related;
    }
}

function Genre(){
    this.name = "";
    this.movies = [];
    this.addMovie = function(m){
        if (m instanceof Movie){
            this.movies.push(m);
            return true;
        }
        return false;
    }
}